src/utils/image.js

```diff
@@ -1,55 +1,106 @@
-      return 'image/tiff';
-  } else if (byteArr[0] === 0x4D && byteArr[1] === 0x4D && byteArr[2] === 0x00 && byteArr[3] === 0x2A) {
-      return 'image/tiff';
+  if (byteArr[0] === 0xff && byteArr[1] === 0xd8 && byteArr[2] === 0xff) {
+    return 'image/jpeg';
+  }
+  if (byteArr[0] === 0x89 && byteArr[1] === 0x50 && byteArr[2] === 0x4e && byteArr[3] === 0x47) {
+    return 'image/png';
+  }
+  if (byteArr[0] === 0x47 && byteArr[1] === 0x49 && byteArr[2] === 0x46) {
+    return 'image/gif';
+  }
+  if (byteArr[0] === 0x42 && byteArr[1] === 0x4d) {
+    return 'image/bmp';
+  }
+  if (byteArr[0] === 0x49 && byteArr[1] === 0x49 && byteArr[2] === 0x2a && byteArr[3] === 0x00) {
+    return 'image/tiff';
+  }
+  if (byteArr[0] === 0x4d && byteArr[1] === 0x4d && byteArr[2] === 0x00 && byteArr[3] === 0x2a) {
+    return 'image/tiff';
   }
 
   // If no known signature is found, return false
   return false;
-}
+};
 
 /**
- * Determines the MIME type of a file based on its extension or base64 string.
- * 
- * @param {string} input - A file extension 
+ * Determines the MIME type of a file based on its source URL/extension or base64 string.
+ *
+ * @param {string} source - A file URL or extension to lookup MIME type from
  * @param {string} base64 - A base64 string representation of the file
- * @returns {string} The MIME type if found, otherwise tries to guess based on base64 content. Returns false if unable to determine.
+ * @returns {string|false} The MIME type if found, otherwise tries to guess based on base64 content. Returns false if unable to determine.
  */
 export const getMimeType = (source, base64) => {
   // Try to lookup the MIME type based on the input directly (assuming it might be a file extension)
   let mimeType = mimeTypes.lookup(source);
 
   // If lookup fails and the input is a base64 string (without the MIME type prefix), try to guess the MIME type
   if (!mimeType && base64 && base64.length > 0) {
-      mimeType = guessMimeTypeFromBase64(base64);
+    mimeType = guessMimeTypeFromBase64(base64);
   }
 
   return mimeType;
-}
\ No newline at end of file
+};
+
+/**
+ * Downloads an image from a URL and converts it to base64.
+ *
+ * @param {string} url - The URL of the image to download
+ * @param {number} timeout - Timeout in milliseconds (default: 5000)
+ * @param {number} maxSize - Maximum allowed size in bytes (default: 10MB)
+ * @returns {Promise<string>} Base64 string of the image
+ * @throws {Error} On timeout, network errors, HTTP errors, or validation failures
+ */
+export async function downloadImageToBase64(url, timeout = 5000, maxSize = 10 * 1024 * 1024) {
+  try {
+    const response = await axios.get(url, {
+      responseType: 'arraybuffer',
+      timeout, // timeout in milliseconds
+      maxContentLength: maxSize, // max size in bytes (10MB default)
+      maxBodyLength: maxSize,
+      validateStatus: (status) => status === 200, // Only accept 200 OK
```

Comment:
AI mentioned the possibility of redirects for image URLs, which will get blocked in these scenarios. Should we avoid these redirects due to potential security reasons, or allow redirects? 🤔